packages/app/src-tauri/src/utils/log.rs

```diff
@@ -0,0 +1,68 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use log::{LevelFilter, Metadata};
+use tauri_plugin_log::LogTarget;
+
+const LOG_LEVEL: LevelFilter = LevelFilter::Error;
+const LOG_TARGETS: [LogTarget; 3] = [LogTarget::Stderr, LogTarget::LogDir, LogTarget::Webview];
+
+#[cfg(not(debug_assertions))]
+static LOG_ENABLED: AtomicBool = AtomicBool::new(false);
+
+#[cfg(debug_assertions)]
+static LOG_ENABLED: AtomicBool = AtomicBool::new(true);
+
+pub fn log_settings() -> (
+    LevelFilter,
+    Vec<LogTarget>,
+    impl Fn(&Metadata<'_>) -> bool + Send + Sync + 'static,
+) {
+    (LOG_LEVEL, Vec::from(LOG_TARGETS), filter_log)
+}
+
+pub fn set_log_status(status: bool) {
+    LOG_ENABLED.store(status, Ordering::Relaxed);
+}
+
+pub fn log_status() -> bool {
+    LOG_ENABLED.load(Ordering::Relaxed)
+}
+
+fn filter_log(_: &Metadata<'_>) -> bool {
+    log_status()
+}
+
+#[macro_export]
+macro_rules! log_if_error {
```

Comment:
maybe a more explicit name cold be helpful, something like `log_if_error_or_return`

Reply:
Thx, renamed as `log_if_error_and_return`